project/hackyourweather-2/src/App.js

```diff
@@ -0,0 +1,58 @@
+import React, { useState } from 'react';
+import AppTitle from './components/AppTitle';
+import CityCard from './components/CityCard';
+import Form from './components/Form';
+import './App.css';
+require('dotenv').config();
+
+function App() {
+  const [weatherInfo, setWeatherInfo] = useState({});
+  const [status, setStatus] = useState('loading');
+  const [hasError, setHasError] = useState('');
+
+  const fetchWeather = async city => {
+    try {
+      const res = await fetch(
+        `https://api.openweathermap.org/data/2.5/weather?q=${city}&units=metric&APPID=${process.env.REACT_APP_OPENWEATHERMAP_API_KEY}`,
+      );
+      if (!res.ok) {
+        throw new Error();
```

Comment:
a simple error message is enough, maybe:
```suggestion
        throw new Error('Oops! Something went wrong.');
```

Or:
```suggestion
        throw new Error('unexpected city name, please try again.');
```
or something you like best, try one that fits your needs and keep it as simple as possible, And that all is for a better user experience. 🙂